services/QuillLMS/spec/queries/quill_big_query/concerns/view_query_fallback_spec.rb

```diff
@@ -2,34 +2,54 @@
 
 require 'rails_helper'
 
-describe QuillBigQuery::MaterializedViewQuery do
+describe QuillBigQuery::Concerns::ViewQueryFallback, type: :module do
   let(:view_key) { 'reporting_sessions_view' }
   let(:name_with_dataset) { 'lms.recent_reporting_sessions_view' }
   let(:name_fallback) { 'recent_reporting_sessions_local' }
   let(:query_fallback) { 'WITH recent_reporting_sessions_local AS (SELECT 2) SELECT 1 FROM recent_reporting_sessions_local' }
   let(:query) { 'SELECT 1 FROM lms.recent_reporting_sessions_view' }
   let(:result) { 'result' }
   let(:runner) { double(:execute) }
+
+  let(:materialized_view) { QuillBigQuery::MaterializedView.new(view_key) }
+
   let(:klass) do
-    Class.new(QuillBigQuery::MaterializedViewQuery) do
-      def materialized_views = [view]
-      def view = materialized_view('reporting_sessions_view')
-      def query = 'SELECT 1 FROM lms.recent_reporting_sessions_view'
+    Class.new do
+      include QuillBigQuery::Concerns::ViewQueryFallback
+
+      define_method(:initialize) do |runner, materialized_view|
+        @runner = runner
+        @materialized_view = materialized_view
+      end
+
+      define_method(:fallback_views) do
+        [@materialized_view]
+      end
+
+      define_method(:runner) do
+        @runner
+      end
+
+      def post_query_transform(result) = result
```

Comment:
Should we use `define_method` here to be consistent with the rest of the methods in this test class?

Reply:
So the `define_method` is only needed for methods that need data from initialize. I've edited the other static method to use the normal `def` setup.